### PR TITLE
main/libgit2: upgrade to 0.26.0; modernize

### PR DESCRIPTION
**Pull request by TBK**
74.4% backward compatibility - https://abi-laboratory.pro/tracker/timeline/libgit2/

### CODE REVIEW

**Comment by ncopa** [comment]
so we probably need to rebuild a bunch of stuff?

**Comment by jirutka** [comment]
Yes.

**Comment by TBK** [comment]
Yup, I know. Working on it :D

**Comment by TBK** [comment]
Julia takes too long to compile and so Travis kills the job.

File: community/julia/APKBUILD [hunk 1]
```diff
@@ -2,10 +2,10 @@
 # Contributor: Shiz <hi@shiz.me>
 # Maintainer: Shiz <hi@shiz.me>
 pkgname=julia
-pkgver=0.6.0
+pkgver=0.6.1
 # Keep in sync with deps/libuv.version.
 _libuv_ver=52d72a52cc7ccd570929990f010ed16e2ec604c8
-_llvm_ver=3.9
+_llvm_ver=4
```
Comment (jirutka) [inline community/julia/APKBUILD]: Julia still recommends LLVM 3.9 according to [README](https://github.com/JuliaLang/julia/blob/v0.6.1/README.md). They are quite sensitive to specific LLVM version, so I’d rather stick to 3.9 until they decide to update.

/cc @yuyichao

**Comment by TBK** [comment]
@jirutka Ok, I have reverted it to 3.9.

**Comment by jirutka** [comment]
@TBK I’ve merged some patches from this PR (aa5d8f79338f88702decde530357b3fa50e22427, d2973e2748c89d3a78e88ef2f3fcd3225dac46a6, 7b9ab3e0f899cf573268eba716417ed817732156). Cargo and Julia needed some more work.

Thanks for your contribution!

**Pull request closed without merging**